### ISSUE TITLE
x64 complains about x86 addons

### ISSUE THREAD

**Issue by alexhass** [notepad-plus-plus/notepad-plus-plus#2410]
### Description of the Issue

Platform mix is not possible.
### Steps to Reproduce the Issue
1. Install x86
2. Install some x86 plugins per user
3. Migrate to x64
4. Start notepad and you get this error.
### Expected Behavior

Helpful message or NO message if platform is different and plugin is not compatible. It may be also useful to place the plugins in an x64 folder to allow users with both notepad platforms to work. e.g. VM with x86 and normal OS with x64. Both with roaming profiles enabled. We cannot make sure that all systems are running the same versions and users may sit on an x86 and x64 system, but their APPDATA is allways the same.
### Actual Behavior

Message annoys users.

![2016-10-09_222346](https://cloud.githubusercontent.com/assets/4450722/19223406/2c4c2126-8e6f-11e6-87cd-ec9972f2e297.png)

**Comment by dail8859** [comment]
Related PR #2348 which provides a better error message.

**Comment by yamlfd** [comment]
Better error messages are wuite useless. I might run both versions on different computers. There should be no message at all.

**Comment by dail8859** [comment]
You expected a "better message" now you say it is useless. 

I never said it was the best solution but for now it is the easiest.

**Comment by yamlfd** [comment]
Yep... some days thinking how stupid error messages for users are. They cannot solve it and do not understsnd it. I expect 100% complains.

Both versions should be able to run in parallel with roaming profiles.

**Comment by yamlfd** [comment]
Any chances to get this SERIOUS bug fixed asap???

**Comment by yamlfd** [comment]
BUMP

**Comment by MAPJe71** [comment]
>Both versions should be able to run in parallel with roaming profiles.
I'm running both 32 and 64 bit Notepad++ sharing one configuration folder (roaming profile). The plugin DLL's are in the applicable `Notepad++\plugins` folder under `Program Files (x86)` and `Program Files` respectively.

**Comment by yamlfd** [comment]
This is not correct! You need to enable per User plugins. Than the files are safed in APPDATA of the user. Than they conflict.

**Comment by MAPJe71** [comment]
>This is not correct!

It's correct, just not your usecase :smile:

>Than they conflict.

As expected. Having both 32 and 64 bit Notepad++ installed with "per user" plugins is not considered a typical usecase so don't expect a solution anytime soon.
Be creative, try to find a work-around e.g. use a batch script which swaps the DLL's prior to starting N++.

**Comment by yamlfd** [comment]
The developer of the software is aware of this feature.

Just move the dlls in a x86/x64 subfolder and all is good. Should be quite simple. Moving to disable is already implemented... so it could be just a few lines of code if you know where.

I'm not running both, but it is possible that larger environments have this. I'd like to switch all our users to x64. From the dll files and with a batch it is not easy to detect what arch they are. This need to be done in the software.

**Comment by MAPJe71** [comment]
>The developer of the software is aware of this feature.

What feature exactly?

>Just move the dlls in a x86/x64 subfolder and all is good.

Not really. 
To run both architectures with plugin's in the user's APPDATA folder (roaming profile) N++ needs be (made) aware the DLL's have moved and where they have been moved to.

>Moving to disable is already implemented...

Yes, in the installer/setup not in N++ itself.

>I'd like to switch all our users to x64.

My bad, I misread your earlier posts as the dual-architecture-with-roaming-profiles being the problem.
Your problem is a failed 32-to-64-bit migration.
Yes, the installer could use some improvement there.

Alternative migration scenario:
1. backup configuration files;
2. uninstall N++ (any installed architecture);
3. install 64 bit N++;
4. optionally restore configuration files. Restore configuration files is optional here as the (un-)installer already has an option to keep the user-data.

**Comment by yamlfd** [comment]
I'm the maintainer of Notepad++ MSI setup, see https://www.hass.de/content/notepad-msi-package-enterprise-deployment-available.

The feature you may have not heard about is placing "allowAppDataPlugins.xml" into the root of "C:\Program Files (x86)\Notepad++". This make notepad saving all user plugins to the users "%APPDATA%\Notepad++\plugins" folder that is a redirected folder on several file servers around the world.

All notepad++ need to do is saving them into "%APPDATA%\Notepad++\plugins\x86" or "%APPDATA%\Notepad++\plugins\x64" or we need to rename the files to take the architecture into consideration.

The setup cannot solve this problem as the setup runs as SYSTEM and NOT in the context of our 500 notepad++ users. The nobrainer part of the NSIS setup that installs something per user is never executed as these folders are not available when setup runs. Users have no permission to install software. I think that notepad++ moves unstable addons to disabled when they crashed, isn't it? It is really years ago when I have experienced this... so not sure.

I cannot run your migration scenarios... this need to be done by the software itself. I cannot run a batch for every user and I do not like to delete the plugin from everyones homes drive. In days where the maintainer removed the plugin manager just because of a stupid banner, this makes it impossible for users to re-install their needed plugins. This is a large enterprise installation... not a home user install.

Nothing against your funny workaround ideas, but they are not working in large scale.

**Comment by dail8859** [comment]
> All notepad++ need to do is saving them into "%APPDATA%\Notepad++\plugins\x86" or "%APPDATA%\Notepad++\plugins\x64" 

If you simple do as you suggest and change where Notepad++ searches for plugins, you have now caused problems for **alot** of users that had existing installations. Since Notepad++ had originally stored them in `%APPDATA%\Notepad++\plugins` directory, then the user would find their updated installation doesn't have any plugins installed (since the new Notepad++ would be looking in a `x86` or `x64` sub directory). This change would also in turn kill off the existing Plugin Manager as it now would not store the plugins in the appropriate locations. (Yes the Plugin Manager could be updated but would cause alot of confusion for users.)

**However,** Notepad++ dropped the Plugin Manager in v7.5 (still usable just no longer shipped in the installer) and is going to implement its own built-in plugin manager (see [this](https://notepad-plus-plus.org/news/notepad-7.5-released.html) page). If this new implementation *does* change how/where plugins are located, then this would be the best time to also try to incorporate a solution to your problem.

**Comment by MAPJe71** [comment]
>The feature you may have not heard about is...

Yes, I'm aware of this "allowAppDataPlugins" feature.

>...that is a redirected folder on several file servers around the world.

Not in general, but at least in your and/or your customers usecase.

> I think that notepad++ moves unstable addons to disabled when they crashed, isn't it?

No, the Notepad++ NSIS setup moves the plugins known to be unstable.

>I'm the maintainer of Notepad++ MSI setup...
>The setup cannot solve this problem as the setup runs as SYSTEM and NOT in the context of our 500 notepad++ users.
>This is a large enterprise installation... not a home user install.

Valuable information to be able to solve your problem.

>Nothing against your funny workaround ideas...

They were not intended to be funny. I was just trying to help (based on the info provided), but glad you liked them.

**Comment by yamlfd** [comment]
> If you simple do as you suggest and change where Notepad++ searches for plugins, you have now caused problems for alot of users that had existing installations. Since Notepad++ had originally stored them in %APPDATA%\Notepad++\plugins directory, then the user would find their updated installation doesn't have any plugins installed (since the new Notepad++ would be looking in a x86 or x64 sub directory). This change would also in turn kill off the existing Plugin Manager as it now would not store the plugins in the appropriate locations. (Yes the Plugin Manager could be updated but would cause alot of confusion for users.)

Maybe you read the previous comments first before commenting? I said notepad need to move the current plugins to these subfolders on first startup. If you can share a different approach (than x86/x64 subfolder) how to solve the problem and allow the installation of both architectures let us know, please.

https://notepad-plus-plus.org/news/notepad-7.5-released.html is not the issue that implements the new plugin manager. Can you share the issue number on github, please?

**Comment by yamlfd** [comment]
> Not in general, but at least in your and/or your customers usecase.
I never said this is the case for everyone. I only say this is something the software MUST expect.

> No, the Notepad++ NSIS setup moves the plugins known to be unstable.
Ok so far. Then we just need a few more lines of code to move the files around.

> They were not intended to be funny. I was just trying to help (based on the info provided), but glad you liked them.
Well there may be some bad workarounds. I could also waste my time, scan all folders for used plugins, download the x64 files manually and replace them on the upgrade days.

But I do not know when people will reboot their computers (and WakeOnLan is not always possible). Than I have no clue when I can replace the plugin files and so on and so on. This must simply fail and still does not allow the use of both x86/x64 versions in parallel. Maybe someone needs a plugin that is only availble in x86, but it is critical for this work. So I cannot migrate the user, but software is always installed per machine. There are only tons of new issues poping up that just need to be handled already by the software.

**Comment by dail8859** [comment]
> I said notepad need to move the current plugins to these subfolders on first startup.

Now knowing this and rereading your messages then yes I see that this is what you were meaning. I interpreted what you were meaning a bit differently (apologies). Since Notepad++ can already determine which plugins are 32 vs 64 bit then in theory this is a possible solution.

> https://notepad-plus-plus.org/news/notepad-7.5-released.html is not the issue that implements the new plugin manager. Can you share the issue number on github, please?

I pointed to that page to show that there is a new plugin manager in progress. There is no issue on Github that I know of. I'm not sure how far this new plugin manager is or how it will function exactly.

**Comment by yamlfd** [comment]
Without an issue on github, how is this coordinated? Every commit should have it's own case.

**Comment by dail8859** [comment]
Currently it is not coordinated because I'm assuming Don Ho is just working on it himself. Pull requests are normally tied to issues but not each commit in the repository is.

**Comment by alexhass** [comment]
We now have October 2018 and x64 Notepad++ still complains about x86 per user plugins. When can we change the folder so this does not conflict?

**Comment by MetaChuh** [comment]
please update your notepad++ to the latest version [<kbd>Notepad++ 7.6.6</kbd>](https://notepad-plus-plus.org/download/v7.6.6.html) and migrate or reinstall your plugins if you update from version 7.5.9 or below.
if your issue still persists, and it is still of relevance, please feel free to file a new issue as easy to understand and reproducible as possible.

thank you for your help and best regards.

**Comment by alexhass** [comment]
This bug is not gixed. Please reopen
